chromium_src/components/policy/tools/template_writers/writer_configuration.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2022 The Brave Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+import override_utils
+
+@override_utils.override_function(globals())
+def GetConfigurationForBuild(original_function, defines):
+  base = original_function(defines)
+  return _merge_dicts(_BRAVE_VALUES, base)
+
+_BRAVE_VALUES = {
+  'build': 'brave',
+  'app_name': 'Brave',
+  'doc_url':
+    'https://support.brave.com/hc/en-us/articles/360039248271-Group-Policy',
+  'frame_name': 'Brave Frame',
+  'webview_name': 'Brave WebView',
+  'win_config': {
+    'win': {
+      'reg_mandatory_key_name': 'Software\\Policies\\BraveSoftware\\Brave',
+      'reg_recommended_key_name':
+        'Software\\Policies\\BraveSoftware\\Brave\\Recommended',
+      'mandatory_category_path': ['Brave:Cat_Brave', 'brave'],
+      'recommended_category_path': ['Brave:Cat_Brave', 'brave_recommended'],
+      'category_path_strings': {
+        'brave': 'Brave',
+        'brave_recommended':
+        'Brave - {doc_recommended}'
+      },
+      'namespace': 'BraveSoftware.Policies.Brave',
+    },
+  },
+  # The string 'Brave' is defined in brave.adml for ADMX, but ADM doesn't
+  # support external references, so we define this map here.
+  'adm_category_path_strings': {
+    'Brave:Cat_Brave': 'Brave'
+  },
+  'admx_prefix': 'brave',
+  'admx_using_namespaces': {
+    'Brave': 'BraveSoftware.Policies'  # prefix: namespace
+  },
+  'linux_policy_path': '/etc/brave/policies/',
+  'bundle_id': 'com.brave.ios.core',
+}
+
+def _merge_dicts(src, dst):
```

Comment:
how about reordering this a little so `GetConfigurationForBuild` goes after `_BRAVE_VALUES` and `_merge_dicts`?
should be easier to graps.

Reply:
I like to organize code in a file in the newspaper style described in [Clean Code](https://www.amazon.de/Clean-Code-Handbook-Software-Craftsmanship/dp/B08X8SDSZ5/ref=sr_1_1?__mk_de_DE=%C3%85M%C3%85%C5%BD%C3%95%C3%91&keywords=clean+code&qid=1667801943&sr=8-1):

> Think of a well-written newspaper article. You read it vertically. At the top you expect a headline that will tell you what the story is about and allows you to decide whether it is something you want to read. The first paragraph gives you a synopsis of the whole story, hiding all the details while giving you the broad-brush concepts. As you continue downward, the details increase until you have all the dates, names, quotes, claims, and other minutia.
> 
> We would like a source file to be like a newspaper article. The name should be simple but explanatory. The name, by itself, should be sufficient to tell us whether we are in the right module or not. The topmost parts of the source file should provide the high-level concepts and algorithms. Detail should increase as we move downward, until at the end we find the lowest level functions and details in the source file.

To me, it reads easier if what's essentially the entry point into the file - `GetConfigurationForBuild` - is at the top. Then I know the high-level flow of the code and can jump to those parts further down I'm interested in.

OK to leave as-is @goodov?

Reply:
function order is OK, but global variable declaration in the middle of other functions doesn't look right to me. I believe most python files has global vars at the top.

Reply:
> I believe most python files has global vars at the top

The [official style guide](https://peps.python.org/pep-0008/) does not say where globals need to go.

I have several issues with this statement:

1. It's dogmatic.
2. It doesn't take into account that "most Python files" may have a reason for putting global vars at the top that doesn't apply here.
3. It's very unspecific. Do we know your statement is true? What percentage of Python files is "most"?

You could make the case that putting globals at the top makes the code easier to read for people who expect this. But the code before the global is so short that I don't think this applies. On the other hand - putting the (long) global before `GetConfigurationForBuild` completely hides the main motivation for having the file. In my view, that makes it strictly harder to understand.

Reply:
> 3\. It's very unspecific. Do we know your statement is true? What percentage of Python files is "most"?

sorry, I'm not skilled enough to quickly calculate the amount of files that have global vars before function/class declaration, so I can't give you the right percentage.

> On the other hand - putting the (long) global before GetConfigurationForBuild completely hides the main motivation for having the file.

should this be a global then?

Reply:
> should this be a global then?

I don't understand the question, sorry. I'd say it already is a global.

> sorry, I'm not skilled enough to quickly calculate the amount of files that have global vars before function/class declaration, so I can't give you the right percentage.

That's fair. I suppose I didn't like the vagueness of "I believe" and "most" and then going on to drawing a conclusion.

Reply:
> On the other hand - putting the **(long) global** before GetConfigurationForBuild completely hides the main motivation for having the file.

should `_BRAVE_VALUES` be global at all or is it better to put it into the overridden function?

Reply:
In my eyes, the structure of `GetConfigurationForBuild` is easier to see when `_BRAVE_VALUES` is a global.